packages/mask/src/extension/popups/pages/Personas/components/ProfileList/index.tsx

```diff
@@ -143,81 +104,34 @@ export const ProfileList = memo(() => {
-        if (!response) return currentPersona.linkedProfiles
-
-        return currentPersona.linkedProfiles.map((profile) => {
-            const target = response.proofs.find(
-                (x) =>
-                    profile.identifier.userId.toLowerCase() === x.identity.toLowerCase() &&
-                    profile.identifier.network.replace('.com', '') === x.platform,
-            )
-
-            return {
-                ...profile,
-                platform: target?.platform,
-                identity: target?.identity,
-                is_valid: target?.is_valid,
-            }
-        })
-    }, [currentPersona])
-
-    const [confirmState, onConfirmDisconnect] = useAsyncFn(async () => {
-        // fetch signature payload
-        try {
-            if (!currentPersona) return
-            const publicHexKey = currentPersona.publicHexKey
-
-            if (!publicHexKey || !unbind || !unbind.identity || !unbind.platform) return
-            const result = await NextIDProof.createPersonaPayload(
-                publicHexKey,
-                NextIDAction.Delete,
-                unbind.identity,
-                unbind.platform,
-            )
-            if (!result) return
-            navigate(
-                urlcat(PopupRoutes.PersonaSignRequest, {
-                    requestID: Math.random().toString().slice(3),
-                    message: result.signPayload,
-                    identifier: currentPersona.identifier.toText(),
-                    method: MethodAfterPersonaSign.DISCONNECT_NEXT_ID,
-                    profileIdentifier: unbind.identifier.toText(),
-                    platform: unbind.platform,
-                    identity: unbind.identity,
-                    createdAt: result.createdAt,
-                    uuid: result.uuid,
-                }),
-            )
-        } catch {
-            console.log('Disconnect failed')
-        }
-    }, [unbind, currentPersona?.identifier, refreshProfileList])
-
     return (
         <>
             <ProfileListUI
-                networks={definedSocialNetworks}
-                profiles={mergedProfiles ?? []}
-                onConnect={onConnect}
+                definedSocialNetworks={props.definedSocialNetworks}
+                mergedProfiles={props.mergedProfiles}
+                onConnectNextID={props.onConnectNextID}
+                onConnectProfile={props.onConnectProfile}
                 onDisconnect={onDisconnect}
-                openProfilePage={openProfilePage}
-            />
-            <DisconnectDialog
-                unbundledIdentity={unbind?.identifier}
-                open={!!unbind}
-                onClose={() => setUnbind(null)}
-                onConfirmDisconnect={onConfirmDisconnect}
-                confirmLoading={confirmState.loading}
+                openProfilePage={props.openProfilePage}
+                SOCIAL_MEDIA_ICON_MAPPING={props.SOCIAL_MEDIA_ICON_MAPPING}
```

Comment:
No need to pass down a constant.

Reply:
I need it because it is defined in `shared` package. If I use it directly, it will pull in the whole `shared` package and bring so many unwanted codes to run.